code/modules/projectiles/pins.dm

```diff
@@ -0,0 +1,204 @@
+/*/////////////////////////
+Alright boys, Firing pins. hopefully with minimal shitcode.
+"pin_auth(mob/living/user)" is the check to see if it fires, put the snowflake code here. return one to fire, zero to flop. ezpz
+
+Firing pins as a rule can't be removed without replacing them, blame a really shitty mechanism for it by NT or something idk, this is to stop people from just taking pins from like a capgun or something.
+*/
+
+
+
+
+
+
+/obj/item/device/firing_pin
+	name = "electronic firing pin"
+	desc = "A small authentication device, to be inserted into a firearm receiver to allow operation. NT safety regulations require all new designs to incorporate one."
+	icon = 'icons/obj/firingpins.dmi'
+	icon_state = "firing_pin"
+	item_state = "pen"
+	origin_tech = "materials=2;combat=4"
```

Comment:
This format is wrong, it should be; origin_tech = list(TECH_MATERIAL = 2, TECH_COMBAT = 4)
Also, why is this giving four of combat?

Reply:
because it's decently hard to get, for starters. I can nerf it down to two if that's what you suggest.